_meta/ecs-migration.yml

```diff
@@ -0,0 +1,148 @@
+#
+# The format of the file is as following:
+#
+# - from: source-field-in-6.x
+#   to: target-filed-in-ECS
+#   # Alias field is useful for fields where there is a 1-1 mapping from old to new
+#   alias: true-if-alias-is-required-in-6x (default is true)
+#   # Copy to is useful for fields where multiple fields map to the same ECS field
+#   copy_to: true-if-field-should-be-copied-to-target-in-6x (default is false)
+
+- from: beat.hostname
+  to: observer.hostname
+
+- from: beat.name
+  to: observer.type
+
+- from: beat.version
+  to: observer.version
+
+- from: context.service.agent.name
+  to: agent.name
+
+- from: context.service.agent.version
+  to: agent.version
+
+- from: context.system.architecture
+  to: host.architecture
+
+- from: context.system.hostname
+  to: host.hostname
+
+- from: context.system.ip
+  to: host.ip
+
+- from: context.system.platform
+  to: host.os.platform
+
+- from: context.request.method
+  to: http.request.method
+
+- from: context.request.http_version
+  to: http.version
+
+- from: context.tags
+  to: labels
+  alias: false
+  copy_to: true
+
+- from: context.process.argv
+  to: process.args
+
+- from: context.process.pid
+  to: process.pid
+
+- from: context.process.ppid
+  to: process.ppid
+
+- from: context.process.title
+  to: process.title
+
+  # not in ECS
+- from: context.service.environment
+  to: service.environment
+
+  # not in ECS
+- from: context.service.framework.name
+  to: service.framework.name
+
+  # not in ECS
+- from: context.service.framework.version
+  to: service.framework.version
+
+  # not in ECS
+- from: context.service.language.name
+  to: service.language.name
+
+  # not in ECS
+- from: context.service.language.version
+  to: service.language.version
+
+- from: context.service.name
+  to: service.name
+
+  # not in ECS
+- from: context.service.runtime.name
+  to: service.runtime.name
+
+  # not in ECS
+- from: context.service.runtime.version
+  to: service.runtime.version
+
+- from: context.service.version
+  to: service.version
+
+- from: context.request.url.full
+  to: url.full
+
+- from: context.request.url.hash
+  to: url.fragment
+
+- from: context.request.url.hostname
+  to: url.domain
+
+- from: context.request.url.pathname
+  to: url.path
+
+- from: context.request.url.port
+  to: url.port
+  alias: false
+  copy_to: true
+
+- from: context.request.url.raw
+  to: url.original
+
+- from: context.request.url.search
+  to: url.query
+
+- from: context.request.url.protocol
+  to: url.scheme
+  alias: false
+  copy_to: true
+
+- from: context.response.finished
+  to: http.response.finished
+
+- from: context.response.status_code
+  to: http.response.status_code
+
+- from: context.user.email
+  to: user.email
+
+- from: context.user.id
+  to: user.id
+
+- from: context.user.username
+  to: user.name
+
+- from: context.user.ip
+  to: client.ip
+
+- from: context.user.user-agent
+  to: user_agent.original.text
+
+- from: listening
+  to: observer.listening
```

Comment:
What's this field meant to contain? There isn't a `observer.listening` in ECS, at this time.

Reply:
This is an awkward field that I'm expecting will never be in ECS.  It's used as a heartbeat so that kibana can verify that an apm-server has checked in, even if it has not processed any traces.

Reply:
Ok. Out of scope for now, but perhaps APM could simply send a completely different (and mostly empty) event that's categorized as a heartbeat? E.g. `event.kind:heartbeat`

Reply:
Agreed, I'll keep an eye on the `event.kind` progress.